### PR TITLE
Move to ppx_expect `v0.17.0`

### PR DESCRIPTION
**Pull request by voodoos**

### CODE REVIEW

**Review by rgrinberg — APPROVED** [review]
I'll update nix in a later PR

**Comment by voodoos** [comment]
Actually `ppx_expect.0.17.0` requires at least OCaml 5.1.

We can either disable testing on 4.14 or wait until we merge 5.2 support.

**Pull request merged**
